overlord/snapstate/target.go

```diff
@@ -754,3 +818,471 @@ func validatedComponentInfo(path string, si *snap.Info, csi *snap.ComponentSideI
 	}
 	return componentInfo, nil
 }
+
+// updatePlan contains the data that describes an update, including a list of
+// Target structs that represent the snaps that are to be updated.
+type updatePlan struct {
+	// requested is the list of snaps that were requested to be updated. If
+	// RefreshAll is true, then this list should be empty.
+	requested []string
+	// targets is the list of snaps that are to be updated. Note that this list
+	// does not necessarily match the list of snaps in Requested.
+	targets []target
+}
+
+// refreshAll returns true if all snaps on the system are being refreshed (could
+// be either an auto-refresh or something like a manual "snap refresh").
+func (p *updatePlan) refreshAll() bool {
+	return len(p.requested) == 0
+}
+
+func (p *updatePlan) targetInfos() []*snap.Info {
```

Comment:
there's a chance in a follow-up that things consuming this should actually be method on updatePlan instead?